confd/templates/logstash/logstash.conf.tmpl

```diff
@@ -63,20 +67,105 @@ input {
   }
 }
 
+
 filter {
   ruby {
     code => "
       event.to_hash.keys.each { |k| event[ k.sub('.','_') ] = event.remove(k) if k.include?'.' }
     "
   }
+
+  ruby {
+    code => "event.set('logstash_processed_at', Time.now());"
+  }
+
   mutate {
       rename => { "logger_name" => "loggerName" }
       rename => { "thread_name" => "thread" }
   }
+
+  mutate {
+    convert => {
+       "timeMillis" => "integer"
+    }
+    tag_on_failure => ["failed_to_convert_timeMillis"]
+  }
+
+  if "failed_to_convert_timeMillis" in [tags] {
+    ruby {
+      code => "event.set('timeMillis', (Time.now.to_f * 1000).to_i)"
+    }
+    mutate {
+      add_field => { "generatedTimeMillis" => "1" }
+    }
+  }
+
+  if "kilda-floodlight" in [tags] {
+    grok {
+      match => { "message" => "%{GREEDYDATA}value:%{GREEDYDATA:floodlight_message}" }
+    }
+    json {
+      source => "floodlight_message"
+      target => "floodlight_message_json"
+      skip_on_invalid_json => true
+    }
+
+    mutate {
+      remove_field => [ "floodlight_message" ]
+    }
+
+    if [floodlight_message_json][payload][timestamp] {
+      ruby {
+        code => "event.set('timeMillis', event.get('[floodlight_message_json][payload][timestamp]').to_i)"
+      }
+    }
+  }
+
+  if ![timeMillis] {
+    if [instant][epochSecond] {
+      if [instant][nanoOfSecond] {
```

Comment:
Is this condition really required?

Reply:
it builds timeMillis from epoch and nanos if it is possible